Add GLM4V (#1668)
huggingface/optimum-habana: examples/image-to-text/README.md

  - [meta-llama/Llama-3.2-90B-Vision-Instruct](https://huggingface.co/meta-llama/Llama-3.2-90B-Vision-Instruct)
  - [tiiuae/falcon-11B-vlm](https://huggingface.co/tiiuae/falcon-11B-vlm)
  - [google/paligemma-3b-mix-224](https://huggingface.co/google/paligemma-3b-mix-224)

>   - [THUDM/glm-4v-9b](https://huggingface.co/THUDM/glm-4v-9b)

Reviewer: we plan to remove this portion also

Author: > we plan to remove this portion also

OK, let me remove it.

Reviewer 2: should be added to the tables in the repo's root README.md and and docs/source/index.mdx

Reviewer: @mengker33 can you please update asap for next release?

Author: Sorry for the late response. I have just updated the doc and fixed the conflicts, please help to review, thank you!

    --sdp_on_bf16
```

> To run THUDM/glm-4v-9b inference, use the following command (Note that we need a env variable "GLM=4v" to distinguish between glm4v and chatglm, this is because these models are all customized models and share the same model_type named "chatglm"):

Reviewer 3: ```suggestion
To run inference with THUDM/glm-4v-9b, use the following command (Note that you need to set the environment variable `GLM=4v` to distinguish between glm4v and chatglm, as these models are customized and share the same model type named "chatglm"):
```

Reviewer 3: @mengker33 GLM-4V requires the Tiktoken library to be installed, please update the requirements.

Reviewer 3: you need to add tiktoken and blobfile packages

Author: Thanks for your comment :) I added tiktoken in image-to-text/requirements, but I didn't find GLM-4V's dependency for blobfile.

Reviewer 3: Please look at https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/src/transformers/utils/import_utils.py#L1202

```sh
def is_tiktoken_available():
    return _tiktoken_available and _blobfile_available
```

    --sdp_on_bf16
```

To run inference with THUDM/glm-4v-9b, use the following command (Note that you need to set the environment variable `GLM=4v` to distinguish between glm4v and chatglm, as these models are customized and share the same model type named "chatglm"):
```bash
GLM=4v python3 run_pipeline.py \
    --model_name_or_path THUDM/glm-4v-9b \
    --use_hpu_graphs \
    --bf16 \
    --sdp_on_bf16 \
    --use_flash_attention \
    --use_kv_cache

> ### Inference with FP8

Reviewer: no need to add in readme

Author: > no need to add in readme

For glm4v, it needs a environment variable GLM=4v to set before running the inference cmd, this is because glm4v shares the same model_type as chatglm defined in their own config.json, and this env var helps to ensure the correct model is registered. I think it necessary to remind user in README of how to run glm4v inference.
